js/apps/system/_admin/aardvark/APP/react/src/views/graphV2/listGraphs/GraphsHelpers.ts

```diff
@@ -0,0 +1,170 @@
+import { GraphInfo } from "arangojs/graph";
+import { getCurrentDB } from "../../../utils/arangoClient";
+import { notifyError, notifySuccess } from "../../../utils/notifications";
+
+export const createGraph = async <
+  ValuesType extends {
+    name: string;
+  }
+>({
+  values,
+  onSuccess
+}: {
+  values: ValuesType;
+  onSuccess: () => void;
+}) => {
+  const currentDB = getCurrentDB();
+  try {
+    const response = await currentDB.request(
+      {
+        method: "POST",
+        path: "/_api/gharial",
+        body: {
+          ...values
+        }
+      },
+      res => res.body.graph
+    );
+    notifySuccess(`Successfully created the graph: ${values.name}`);
+    onSuccess();
+    return response;
+  } catch (e: any) {
+    const errorMessage = e.response.body.errorMessage;
+    notifyError(`Could not add graph: ${errorMessage}`);
+    return {
+      error: errorMessage
+    };
+  }
+};
+
+type DetectedGraphType = "satellite" | "enterprise" | "smart" | "general";
+
+const GRAPH_TYPE_TO_LABEL_MAP = {
+  satellite: "Satellite",
+  smart: "Smart",
+  enterprise: "Enterprise",
+  general: "General"
+};
+
+const getGraphLabel = ({
+  type,
+  isDisjoint
+}: {
+  type: DetectedGraphType;
+  isDisjoint?: boolean;
+}) => {
+  let graphTypeString = GRAPH_TYPE_TO_LABEL_MAP[type];
+  if (isDisjoint) {
+    return `Disjoint ${graphTypeString}`;
+  }
+  return graphTypeString;
+};
+
+export const detectGraphType = (
+  graph: GraphInfo
+): {
+  type: DetectedGraphType;
+  isDisjoint?: boolean;
+  label: string;
+} => {
+  if (graph.isSatellite || (graph.replicationFactor as any) === "satellite") {
+    return {
+      type: "satellite",
+      label: getGraphLabel({ type: "satellite" })
+    };
+  }
+
+  if (graph.isSmart && graph.smartGraphAttribute === undefined) {
```

Comment:
This way could catch other falsy values:
```suggestion
  if (graph.isSmart && !graph.smartGraphAttribute) {
```